web/controllers/v1/bundle_version_controller.ex

```diff
@@ -12,23 +12,31 @@ defmodule Cog.V1.BundleVersionController do
       %BundleVersion{}=bundle_version ->
         render(conn, "show.json", %{bundle_version: bundle_version})
       nil ->
-        send_resp(conn, 404, Poison.encode!(%{error: "Bundle #{id} not found"}))
+        conn
+        |> put_status(:not_found)
+        |> json(%{error: "Bundle version #{id} not found"})
     end
   end
 
-  def delete(conn, %{"id" => id}) do
+  def delete(conn, %{"bundle_version_id" => id}) do
     case Repository.Bundles.version(id) do
       %BundleVersion{}=bv ->
         case Repository.Bundles.delete(bv) do
           {:ok, _} ->
             send_resp(conn, 204, "")
```

Comment:
Phoenix does some automatic response handling if the controller code throws an exception, provided you've supplied the appropriate exception -> status code mappings. I don't think that applies here.